src/category_theory/punit.lean

```diff
@@ -70,4 +71,25 @@ def equiv : (discrete punit ⥤ C) ≌ C :=
 
 end functor
 
+/-- A category being equivalent to `punit` is equivalent to it having a unique morphism between
+  any two objects. (In fact, such a category is also a groupoid; see `groupoid.of_all_unique`) -/
+theorem equiv_punit_iff_unique :
+  nonempty (C ≌ discrete punit) ↔ (nonempty C) ∧ (∀ x y : C, nonempty $ unique (x ⟶ y)) :=
+begin
+  split,
+  { rintro ⟨h⟩, refine ⟨⟨h.inverse.obj punit.star⟩, λ x y, nonempty.intro _⟩,
+    apply (unique_of_subsingleton _), swap,
+    { have hx : x ⟶ h.inverse.obj punit.star := by convert h.unit.app x,
+      have hy : h.inverse.obj punit.star ⟶ y := by convert h.unit_inv.app y,
+      exact hx ≫ hy, },
+    suffices : ∀ z, z = h.unit.app x ≫ (h.functor ⋙ h.inverse).map z ≫ h.unit_inv.app y,
+    { apply subsingleton.intro, intros a b, rw [this a, this b],
```

Comment:
```suggestion
    { apply subsingleton.intro, 
      intros a b, 
      rw [this a, this b],
```
Lines are cheap. Typically only put multiple tactics on one line if
* it means you can fit the whole proof in one line
* you want to signify that the tactics "go together" somehow as a single step (e.g. `use x, clear x`)

Reply:
(The `apply ... swap` above is a good example where two tactics on one line is helpful to the reader.)